tests/Builder/BuilderEncoderTest.php

```diff
@@ -342,26 +340,11 @@ private static function assertSamePipeline(array $expected, Pipeline $pipeline):
         $codec = new BuilderEncoder();
         $actual = $codec->encode($pipeline);
 
-        self::objectify($expected);
+        // Normalize with BSON round-trip
+        // BSON Documents doesn't support top-level arrays.
+        $actual = Document::fromPHP(['root' => $actual])->toRelaxedExtendedJSON();
```

Comment:
Any reason not to use [`toCanonicalExtendedJSON()`](https://www.php.net/manual/en/mongodb-bson-document.tocanonicalextendedjson.php)?

Reply:
I must admit that I'm very confused between the 2 methods. `toCanonicalExtendedJSON()` to be the correct one to use.